contracts/CTKNCrowdsale.sol

```diff
@@ -138,8 +177,36 @@ contract CTKNCrowdsale is CappedCrowdsale, RefundableCrowdsale, MintedCrowdsale
      *  @param _weiAmount Value in wei to be converted into tokens
      *  @return the wei amount in excess of what was needed to buy a whole number of tokens.
      */
-    function _getOverpaymentAmount(uint256 _weiAmount) internal view returns (uint256) {
+    function _getOverpaymentAmount(uint256 _weiAmount)
+        internal
+        view
+        returns (uint256)
+    {
       return _weiAmount % rate;
     }
 
+    /**
+     * Overrised parent behavior requiring purchase to respect the funding cap but in USD
+     * @param _beneficiary Token purchaser
+     * @param _weiAmount Amount of wei contributed
+     */
+    function _preValidatePurchase(address _beneficiary, uint256 _weiAmount)
```

Comment:
Also, not sure if working with KYC we can use WhitelistedCrowdsale: https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/crowdsale/validation/WhitelistedCrowdsale.sol and override that _preValidatedPurchase . That way we can have "Shoppers" be whitelisted into investors per @smoliver diagram.

Reply:
In our case the whitelist is attached to the token and not the crowdsale (or maybe it needs to be attached to both?)

Reply:
either way that can happen in a future PR